spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/data/redis/RedisAutoConfigurationLettuceUrlTests.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2012-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.data.redis;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.data.redis.connection.RedisSentinelConfiguration;
+import org.springframework.data.redis.connection.RedisSocketConfiguration;
+import org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests {@link RedisAutoConfiguration} when using {@code Lettuce} and configuring via the
+ * {@code spring.redis.url} property.
+ *
+ * @author Chris Bono
+ */
+class RedisAutoConfigurationLettuceUrlTests {
```

Comment:
I chose to break these out into their own test as the existing test was already pretty unwieldy and adding this in would not help. If this is an issue we can of course consolidate back in.

Reply:
There isn't a relationship between the main test and the jedis related one. If you want to break things down, why not `LettuceConnectionConfigurationTests` ?

Reply:
I guess naming this new one containing the words "AutoConfiguration" and "Lettuce" may make it seem like this new test is the one that tests the Lettuce variant of auto-config when really the main test `RedisAutoConfigurationTests` does that. 

Naming this new test as you recommend is a bit clearer and may produce less confusion. I will adjust.

Reply:
Ok after going back into the code I think the current name makes sense. Let me summarize my thinking and you can tell me if I am off of what you are thinking about this @snicoll.


#### Before my changes: 

`RedisAutoConfiguration`: 
- tests focused on provider agnostic auto-config features (albeit w/ Lettuce being the default provider). 
- handles the URL configured cases (with default provider)

`RedisAutoConfigurationJedisTests`: 
- tests when Jedis is used as the provider
- not as exhaustive as the main `RedisAutoConfiguration` test and only focuses on URL configured cases and other Jedis specifics.

At this point the URL configured cases are handled for both providers from the 2 above tests.

#### My changes:

Rather than put the new Lettuce specific URL configured cases in `RedisAutoConfiguration` I split it out into its own Lettuce URL auto-config test and removed any URL configured cases from the main test.

Up to this point the `LettuceConnectionConfiguration` and `JedisConnectionConfiguration` are tested implicitly via the _RedisAutoConfiguration*Tests_. I think naming this Lettuce URL test `LettuceConnectionConfigurationTests` will leave contributors wondering where the Jedis version is and also it would not be testing all of the LettuceConnectionConfiguration. Therefore I think leaving it named as `RedisAutoConfigurationLettuceUrlTests` makes sense.